# **Random Forest 예제**

## 1. 데이터 로드 
###데이터 세트
* Personal Key Indicators of Heart Disease  
(2020 annual CDC survey data of 400k adults related to their health status)  
* Source : https://www.cdc.gov/brfss/annual_data/annual_2020.html
* Offer : Kaggle
* License : Public Domain
* Collaborators : Kamil Pytlak

In [29]:
#데이터 로드
################################################################################
import pandas as pd
url = 'https://drive.google.com/file/d/1FS49mw-RqhCAp5T7rE6q5FiuTId7o8_9/view?usp=sharing'
url = 'https://drive.google.com/uc?id=' + url.split('/')[-2]
csvData = pd.read_csv(url)
csvData.head()

HeartDisease    BMI Smoking AlcoholDrinking Stroke  PhysicalHealth  \
0           No  16.60     Yes              No     No             3.0   
1           No  20.34      No              No    Yes             0.0   
2           No  26.58     Yes              No     No            20.0   
3           No  24.21      No              No     No             0.0   
4           No  23.71      No              No     No            28.0   

   MentalHealth DiffWalking     Sex  AgeCategory   Race Diabetic  \
0          30.0          No  Female        55-59  White      Yes   
1           0.0          No  Female  80 or older  White       No   
2          30.0          No    Male        65-69  White      Yes   
3           0.0          No  Female        75-79  White       No   
4           0.0         Yes  Female        40-44  White       No   

  PhysicalActivity  GenHealth  SleepTime Asthma KidneyDisease SkinCancer  
0              Yes  Very good        5.0    Yes            No        Yes  
1              Yes  Very good        7.0     No            No         No  
2              Yes       Fair        8.0    Yes            No         No  
3               No       Good        6.0     No            No        Yes  
4              Yes  Very good        8.0     No            No         No

In [22]:
#데이터 헤더
################################################################################
#csvData.data.shape
list(csvData)

['HeartDisease',
 'BMI',
 'Smoking',
 'AlcoholDrinking',
 'Stroke',
 'PhysicalHealth',
 'MentalHealth',
 'DiffWalking',
 'Sex',
 'AgeCategory',
 'Race',
 'Diabetic',
 'PhysicalActivity',
 'GenHealth',
 'SleepTime',
 'Asthma',
 'KidneyDisease',
 'SkinCancer']

In [41]:
#데이터 형태
################################################################################
nRow = csvData.shape[0] # 데이터 개수
nCol = csvData.shape[1] # 변수 개수
print('nRow: %d' % nRow, '/ nCol: %d' % nCol )
csvData.info()

nRow: 319795 / nCol: 18
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 319795 entries, 0 to 319794
Data columns (total 18 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   HeartDisease      319795 non-null  object 
 1   BMI               319795 non-null  float64
 2   Smoking           319795 non-null  object 
 3   AlcoholDrinking   319795 non-null  object 
 4   Stroke            319795 non-null  object 
 5   PhysicalHealth    319795 non-null  float64
 6   MentalHealth      319795 non-null  float64
 7   DiffWalking       319795 non-null  object 
 8   Sex               319795 non-null  object 
 9   AgeCategory       319795 non-null  object 
 10  Race              319795 non-null  object 
 11  Diabetic          319795 non-null  object 
 12  PhysicalActivity  319795 non-null  object 
 13  GenHealth         319795 non-null  object 
 14  SleepTime         319795 non-null  float64
 15  Asthma            319795 non-null  object 
 

###데이터 구조
*   데이터 개수  : 319795 개 (헤더 제외)  
*   변수 개수: 18 개

###설명 변수 (17개)
1. BMI : 체질량 지수
2. Smoking : 흡연
3. AlcoholDrinking : 음주
4. Stroke : 뇌졸중
5. PhysicalHealth : 육체적 건강
6. MentalHealth : 정신적 건강
7. DiffWalking : 걷기 장애
8. Sex : 성별
9. AgeCategory : 나이대
10. Race : 인종
11. Diabetic : 당뇨
12. PhysicalActivity : 신체 활동
13. GenHealth : ?
14. SleepTime : 수면 시간
15. Asthma : 소아 천식
16. KidneyDisease : 신장 질환
17. SkinCancer : 피부암

###반응 변수 (1개)
1.   HeartDisease : 심장병

